nvflare/private/fed/server/job_validator.py

```diff
@@ -0,0 +1,150 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import collections
+import json
+import time
+from datetime import datetime
+from io import BytesIO
+from typing import Set, Tuple
+from zipfile import ZipFile
+
+from nvflare.apis.fl_constant import SystemComponents
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.job_def import JobMetaKey
+from nvflare.apis.study_manager_spec import Study
+
+
+class JobValidator:
+    """Job validator"""
+
+    def __init__(self, fl_ctx: FLContext):
+        self.fl_ctx = fl_ctx
+        engine = fl_ctx.get_engine()
+        self.study_manager = engine.get_component(SystemComponents.STUDY_MANAGER)
+        if not self.study_manager:
+            raise ValueError("Server configuration error: no study manager is configured")
+
+    def validate(self, job_name: str, job_data: bytes) -> Tuple[bool, str, dict]:
+        """Validate job
+
+        Args:
+            job_name (str): Job name
+            job_data (bytes): Job ZIP data
+
+        Returns:
+            Tuple[bool, str, dict]: (is_valid, error_message, meta)
+        """
+
+        meta = {}
+        try:
+            with ZipFile(BytesIO(job_data), "r") as zf:
+                meta_file = job_name + "/meta.json"
+                meta_data = zf.read(meta_file)
```

Comment:
If meta_file does not exist, what will happen?

Reply:
One will be created if meta.json is missing.

Reply:
the read method won't create file, correct ? are you sure the meta.json not exists, it will auto create one ?